jcl/src/openj9.criu/share/classes/org/eclipse/openj9/criu/CRIUSupport.java

```diff
@@ -114,43 +119,160 @@ public static boolean isCRIUSupportEnabled() {
 		return criuSupportEnabled;
 	}
 
+	private String imagesDir;
+	private boolean leaveRunning;
+	private boolean shellJob;
+	private boolean extUnixSupport;
+	private int logLevel;
+	private String logFile;
+	private boolean fileLocks;
+	private String workDir;
+
+	/**
+	 * Sets the directory that will hold the images upon dump. Corresponds to criu_set_images_dir_fd.
+	 * This must be set before calling {@link #checkPointJVM()}.
+	 * 
+	 * @param imagesDir the directory as a java.nio.file.Path
+	 * @return this
+	 * @throws NullPointerException if imagesDir is null.
+	 * @throws SecurityException if no permission to access imagesDir.
+	 */
+	public CRIUSupport setImagesDir(Path imagesDir) {
+		Objects.requireNonNull(imagesDir, "Path cannot be null");
+		String dir = imagesDir.toAbsolutePath().toString();
+
+		SecurityManager manager = System.getSecurityManager();
+		if (manager != null) {
+			manager.checkPermission(CRIU_DUMP_PERMISSION);
```

Comment:
The dump permission check only needs to be done once, so that can be done in the constructor